internal/tofu/eval_context_builtin.go

```diff
@@ -155,21 +155,6 @@ func (ctx *BuiltinEvalContext) Provider(addr addrs.AbsProviderConfig) providers.
 }
 
 func (ctx *BuiltinEvalContext) ProviderSchema(addr addrs.AbsProviderConfig) (providers.ProviderSchema, error) {
-	// first see if we have already have an initialized provider to avoid
-	// re-loading it only for the schema
-	p := ctx.Provider(addr)
-	if p != nil {
-		resp := p.GetProviderSchema()
-		// convert any diagnostics here in case this is the first call
-		// FIXME: better control provider instantiation so we can be sure this
-		// won't be the first call to ProviderSchema
-		var err error
-		if resp.Diagnostics.HasErrors() {
-			err = resp.Diagnostics.ErrWithWarnings()
-		}
-		return resp, err
-	}
```

Comment:
Could you expand on why we're removing this?

Isn't this an optimization not to start new provider instances if ones already exist?

Reply:
Without this, won't we always create a new instance?

Reply:
ctx.Plugins already interfaces with the provider schema cache.  That cache is "global" and populated any time any context starts an instance of the corresponding type.  Additionally, ctx.Plugins provides additional validation that this may short circuit.

This is an older abstraction which is made obsolete by some of the fixes made to provider schema caching.

Reply:
Alright